pkg/api/customization/authn/user_store.go

```diff
@@ -188,3 +188,43 @@ func (s *userStore) create(apiContext *types.APIContext, schema *types.Schema, d
 
 	return s.Store.Create(apiContext, schema, data)
 }
+
+func (s *userStore) Update(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, id string) (map[string]interface{}, error) {
+	currentUser, err := getUser(apiContext)
+	if err != nil {
+		return nil, err
+	}
+
+	willBeInactive := false
+	if val, ok := data[client.UserFieldEnabled].(bool); ok {
+		willBeInactive = !val
+	}
+
+	if currentUser == id && willBeInactive {
+		return nil, errors.New("You cannot deactivate yourself")
+	}
+
+	return s.Store.Update(apiContext, schema, data, id)
+}
+
+func (s *userStore) Delete(apiContext *types.APIContext, schema *types.Schema, id string) (map[string]interface{}, error) {
+	currentUser, err := getUser(apiContext)
+	if err != nil {
+		return nil, err
+	}
+
+	if currentUser == id {
+		return nil, errors.New("You cannot delete yourself")
+	}
+
+	return s.Store.Delete(apiContext, schema, id)
+}
+
+func getUser(apiContext *types.APIContext) (string, error) {
+	user := apiContext.Request.Header.Get("Impersonate-User")
+	if user == "" {
+		return "", httperror.NewAPIError(httperror.NotFound, "missing user")
```

Comment:
If this header is missing at this point, rancher has royally screwed up. Can you change the error code to 500 and a message/code indicating that there was an error authorizing the request.
Its not a 401 or 403 because this is a bug if we get to this point in the code and that header is not present